examples/cli/readme.md

```diff
@@ -1 +1,619 @@
+
+You can try this out with the following command:
+
+```
+benchit benchmark scripts/hello_world.py --cache-dir tmp_cache
+```
+
+When that command completes, you can use the `ls` command to see that `tmp_cache` has been created at your command line location. 
+
+See the [Cache Commands](#cache-commands) tutorial to see what you can do with the cache.
+
+> See the [Cache Directory documentation](https://github.com/groq/mlagility/blob/main/tools_user_guide.md#cache-directory) for more details.
+
+## Cache Commands
+
+This tutorial assumes you have completed the [Cache Directory](#cache-directory) and [Benchmark All Scripts](#benchmark-all-scripts) tutorials, and that the `tmp_cache` directory exists at your command line location.
+
+You can use the `cache list` command to see what builds are available in your cache:
+
+```
+benchit cache list
+```
+
+Which produces a result like:
+
+```
+Info: Builds available in cache ~/.cache/mlagility:
+hello_world_479b1332     max_depth_80b93950       two_models_479b1332      
+        labels                   two_models_215ca1e3 
+```
+
+> _Note_: `labels` is not a build, it is a helper directory to store labeling data.
+
+You can learn more about a build with the `cache stats` command:
+
+```
+benchit cache stats hello_world_479b1332
+```
+
+Which will print out a lot of statistics about the build, like:
+
+```
+Info: The state of build hello_world_479b1332 in cache ~/.cache/mlagility is:
+build_status: successful_build
+cache_dir: /home/jfowers/.cache/mlagility
+config:
+  assembler_flags:
+  - --ifetch-from-self
+  - --ifetch-slice-ordering=round-robin
+  build_name: hello_world_479b1332
+  compiler_flags: []
+  groqcard: A1.4
+  groqview: false
+  num_chips: null
+  sequence:
+  - export_pytorch
+  - optimize_onnx
+  - fp16_conversion
+  - set_success
+...
+```
+
+You can also delete a build from a cache with the `cache delete` command. Be careful, this permanently deletes the build!
+
+For example, you could run the commands:
+
+```
+benchit cache delete max_depth_80b93950
+benchit cache list
+```
+
+And you would see that the cache no longer includes the build for `max_depth`:
+
+```
+Info: Builds available in cache ~/.cache/mlagility:
+hello_world_479b1332     labels                   two_models_215ca1e3      two_models_479b1332 
+```
+
+Finally, the `cache` commands all take a `--cache-dir` that allows them to operate on a specific cache directory (see the [Cache Directory tutorial](#cache-directory) for more details).
+
+For example, you can run this command:
+
+```
+benchit cache list --cache-dir tmp_cache
+```
+
+Which will produce this result, if you did the [Cache Directory tutorial](#cache-directory):
+
+```
+Info: Builds available in cache tmp_cache:
+hello_world_479b1332     labels  
+```
+
+> See the [Cache Commands documentation](https://github.com/groq/mlagility/blob/main/tools_user_guide.md#cache-commands) for more details.
+
+## Lean Cache
```

Comment:
Reading this makes me think we should potentially have a "lean" command that makes a large cache "lean".

Reply:
`benchit cache delete --all --artifacts`?

Reply:
Where `--artifacts` specifies that only artifacts should be deleted from the build

Reply:
Something along those lines. If we add `--artifacts` then I would rename `--lean-cache` to `--no-artifacts `

Reply:
Added #128